### PR TITLE
Ots 743

### PR DESCRIPTION
**Pull request by southfox**
## Pull request checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I have resolved any merge conflicts. Confirmation: ____

## What's in this pull request?

>AppDelegate: create an acceleratorSession instance that is shared inside the example application.
> Fixing project.pbxproj not included in the repository.
> Using OTAcceleratorPackUtil ~> 2.0.0-beta4

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/opentok/one-to-one-sample-apps/pull/161" target="_blank">ReviewNinja

**Comment by southfox** [comment]
Replaces PR https://github.com/opentok/one-to-one-sample-apps/pull/159

**Comment by Lucashuang0802** [comment]
As we handle audio/video on/off on the acc-pack level, we can remove the placeholder image from app level. `showAudioVideoControl` should be set to `NO` because we don't need the default UI for handle audio/video. Reading internal codes might give more intent.

File: iOS/SampleApp/MainView.m [hunk 1]
```diff
@@ -178,12 +140,11 @@ - (void)showReverseCameraButton; {
 
 - (void)resetAllControl {
     [self removePublisherView];
-    [self removePlaceHolderImage];
     [self connectCallHolder:NO];
     [self updatePublisherAudio:YES];
     [self updatePublisherVideo:YES];
     [self updateSubscriberAudioButton:YES];
-    [self updateSubsciberVideoButton:YES];
+    [self updateSubscriberVideoButton:YES];
```
Comment (Lucashuang0802) [inline iOS/SampleApp/MainView.m]: 👍

File: iOS/SampleApp/Info.plist [hunk 2]
```diff
@@ -26,6 +22,10 @@
 	<string>1</string>
 	<key>LSRequiresIPhoneOS</key>
 	<true/>
+	<key>NSCameraUsageDescription</key>
+	<string>$(PRODUCT_NAME) uses camera</string>
+	<key>NSMicrophoneUsageDescription</key>
+	<string>$(PRODUCT_NAME) uses microphone</string>
```
Comment (Lucashuang0802) [inline iOS/SampleApp/Info.plist]: We don't need this change right?

File: iOS/SampleApp/AppDelegate.m [hunk 3]
```diff
@@ -45,4 +47,12 @@ - (void)applicationWillTerminate:(UIApplication *)application {
   // Saves changes in the application's managed object context before the application terminates.
 }
 
+- (OTAcceleratorSession *)acceleratorSession {
+    if (!_acceleratorSession) {
+        _acceleratorSession = [[OTAcceleratorSession alloc] initWithOpenTokApiKey:kApiKey
```
Comment (Lucashuang0802) [inline iOS/SampleApp/AppDelegate.m]: You can put it back to `didFinishLaunchingWithOptions`. it's not necessary to do this way.

File: iOS/SampleApp/AppDelegate.m [hunk 4]
```diff
@@ -5,20 +5,22 @@
 //
 
 #import "AppDelegate.h"
-#import <OTAcceleratorPackUtil/OTAcceleratorPackUtil.h>
+#import <OTAcceleratorPackUtil/OTAcceleratorSession.h>
 
-@interface AppDelegate ()
+#define kApiKey @"<# Replace #>"
+#define kSessionId @"<# Replace #>"
+#define kToken @"<# Replace #>"
```
Comment (Lucashuang0802) [inline iOS/SampleApp/AppDelegate.m]: Let's not use MARCO as an example.

**Comment by Lucashuang0802** [comment]
👍 just need to be careful as it's an example to provide to developers.

**Comment by southfox** [comment]
Thanks @Lucashuang0802, I've updated the PR with your suggestions.

File: iOS/SampleApp/MainViewController.m [hunk 5]
```diff
@@ -75,53 +89,41 @@ - (void)handleCommunicationSignal:(OTOneToOneCommunicationSignal)signal {
...
-            [SVProgressHUD showErrorWithStatus:@"Problem when subscribing."];
             break;
         }
         case OTSubscriberVideoDisabledByBadQuality:
         case OTSubscriberVideoDisabledBySubscriber:
-        case OTSubscriberVideoDisabledByPublisher:{
-            [self.mainView removeSubscriberView];
-            [self.mainView addPlaceHolderToSubscriberView];
+        case OTSubscriberVideoDisabledByPublisher:
+        {
```
Comment (Lucashuang0802) [inline iOS/SampleApp/MainViewController.m]: Let's align to the previous line.

File: iOS/SampleApp/MainViewController.m [hunk 6]
```diff
@@ -21,20 +25,30 @@ - (void)viewDidLoad {
     [super viewDidLoad];
     
     self.mainView = (MainView *)self.view;
-    self.oneToOneCommunicator = [OTOneToOneCommunicator sharedInstance];
+    
+    MAKE_WEAK(self);
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(1.0 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
```
Comment (Lucashuang0802) [inline iOS/SampleApp/MainViewController.m]: Can we try to avoid this pattern?

File: iOS/SampleApp/MainViewController.m [hunk 7]
```diff
@@ -21,20 +25,30 @@ - (void)viewDidLoad {
...
     
-    [SVProgressHUD show];
-    
     if (!self.oneToOneCommunicator.isCallEnabled) {
+        [SVProgressHUD show];
+
+        MAKE_WEAK(self);
         [self.oneToOneCommunicator connectWithHandler:^(OTOneToOneCommunicationSignal signal, NSError *error) {
+            MAKE_STRONG(self);
+            strongself.oneToOneCommunicator.publisherView.showAudioVideoControl = NO;
```
Comment (Lucashuang0802) [inline iOS/SampleApp/MainViewController.m]: 👍

File: iOS/SampleApp/MainViewController.m (hunk 6 above)
Comment (Lucashuang0802) [inline iOS/SampleApp/MainViewController.m]: I mean the `dispatch_after`. perhaps

**Comment by southfox** [comment]
Thanks @marinaserranomontes for the fix, and it's ready for you to review now @Lucashuang0802.

File: iOS/.gitignore [hunk 8]
```diff
@@ -21,5 +21,5 @@ Pods
 
 *.framework
 *.bundle
+Podfile.lock
```
Comment (Lucashuang0802) [inline iOS/.gitignore]: Do not commit `.lock` file.

File: iOS/SampleApp/AppDelegate.m [hunk 9]
```diff
@@ -5,21 +5,20 @@
 //
 
 #import "AppDelegate.h"
-#import <OTAcceleratorPackUtil/OTAcceleratorPackUtil.h>
+#import <OTAcceleratorPackUtil/OTAcceleratorSession.h>
```
Comment (Lucashuang0802) [inline iOS/SampleApp/AppDelegate.m]: You don't actually need `<` and `>`. it's completed opened. using double quotes should be fine.

Comment (Lucashuang0802) [inline iOS/SampleApp/AppDelegate.m]: You don't actually need `<` and `>`. it's completed opened. using double quotes should be fine.

File: iOS/.gitignore (hunk 8 above)
Comment (southfox) [inline iOS/.gitignore]: Is in .gitignore, do you want me to delete the line there?

Comment (Lucashuang0802) [inline iOS/.gitignore]: totoally my bad! :+1:

File: iOS/SampleApp/AppDelegate.m [hunk 10]
```diff
@@ -5,21 +5,20 @@
 //
 
 #import "AppDelegate.h"
-#import <OTAcceleratorPackUtil/OTAcceleratorPackUtil.h>
+#import "OTAcceleratorSession.h"
```
Comment (Lucashuang0802) [inline iOS/SampleApp/AppDelegate.m]: 👍

**Comment by Lucashuang0802** [comment]
👍

**Pull request merged**